### PR TITLE
Release v0.3 

### PR DESCRIPTION
**Pull request by Kostronor**
Pull request to release the newest changes with achievements.

Will be playtested before merge!

### CODE REVIEW

File: src/main/java/com/minecolonies/client/gui/WindowTownHall.java [hunk 1]
```diff
@@ -314,6 +467,85 @@ public void updateElement(int index, Pane rowPane)
...
+     * Fills the workOrder list inside the townhall GUI.
+     */
+    private void fillWorkOrderList()
+    {
+        final ScrollingList workOrderList =  findPaneOfTypeByID(LIST_WORKORDER, ScrollingList.class);
+        workOrderList.enable();
+        workOrderList.show();
+
+        //Creates a dataProvider for the unemployed citizenList.
+        workOrderList.setDataProvider(new ScrollingList.DataProvider()
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/client/gui/WindowTownHall.java]: MAJOR Reduce this anonymous class number of lines from 45 to at most 30, or make it a named class. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1188)

File: src/main/java/com/minecolonies/colony/Colony.java [hunk 2]
```diff
@@ -519,16 +537,42 @@ private void sendCitizenPackets(Set<EntityPlayerMP> oldSubscribers, boolean hasN
...
+     * Sends packages to update the workOrders.
+     *
+     * @param oldSubscribers    the existing subscribers.
+     * @param hasNewSubscribers the new subscribers.
+     */
+    private void sendWorkOrderPackets(Set<EntityPlayerMP> oldSubscribers, boolean hasNewSubscribers)
+    {
+        if (getWorkManager().isDirty() || hasNewSubscribers)
+        {
+            for (AbstractWorkOrder workOrder : getWorkManager().getWorkOrders().values())
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/colony/Colony.java]: MINOR Local variable 'workOrder' could be declared final [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)

File: src/main/java/com/minecolonies/colony/Colony.java [hunk 3]
```diff
@@ -904,20 +981,34 @@ public void removeCitizen(CitizenData citizen)
...
+     * Send the message of a removed workOrder to the client.
      *
-     * @param citizenId ID of the Citizen
-     * @return          CitizenData associated with the ID, or null if it was not found
+     * @param orderId the workOrder to remove.
+     */
+    public void removeWorkOrder(int orderId)
+    {
+        //  Inform Subscribers of removed workOrder
+        for (EntityPlayerMP player : subscribers)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/colony/Colony.java]: MINOR Local variable 'player' could be declared final [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)

File: src/main/java/com/minecolonies/colony/WorkManager.java [hunk 4]
```diff
@@ -211,13 +231,39 @@ public void onWorldTick(TickEvent.WorldTickEvent event)
                 if (!o.isValid(colony))
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/colony/WorkManager.java]: MAJOR Avoid if (x != y) ..; else ..; [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AConfusingTernary)

File: src/main/java/com/minecolonies/proxy/ClientProxy.java [hunk 5]
```diff
@@ -67,6 +68,10 @@ public void showCitizenWindow(CitizenDataView citizen)
     @Override
     public void openBuildToolWindow(BlockPos pos)
     {
+        if (pos == null && Settings.instance.getActiveSchematic() == null) {
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/proxy/ClientProxy.java]: MINOR Move this left curly brace to the beginning of next line of code. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

File: src/main/java/com/minecolonies/colony/jobs/JobBuilder.java [hunk 6]
```diff
@@ -159,20 +168,27 @@ public void setSchematic(SchematicWrapper schematic)
...
-     * Returns the work interval of the worker //unfinished
+     * Returns the work interval of the worker.
      *
      * @return work interval
      */
     public int getWorkInterval()
     {
-        return 1;//Constants.BUILDERWORKINTERFALL - this.getLevel();//TODO
+        //TODO
+        //Constants.BUILDERWORKINTERFALL - this.getLevel();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/colony/jobs/JobBuilder.java]: MAJOR This block of commented-out lines of code should be removed. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: src/main/java/com/minecolonies/items/ItemBuildTool.java [hunk 7]
```diff
@@ -22,36 +24,25 @@ public String getName()
     }
 
     @Override
+    public ItemStack onItemRightClick(ItemStack stack, World world, EntityPlayer player) {
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/items/ItemBuildTool.java]: MINOR Move this left curly brace to the beginning of next line of code. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

File: src/main/java/com/minecolonies/colony/workorders/WorkOrderBuild.java [hunk 8]
```diff
@@ -191,9 +200,16 @@ public void attemptToFulfill(Colony colony)
...
      */
-    private boolean extraChecks(int builderLevel)
+    private boolean canBuildHut(int builderLevel, CitizenData citizen, Colony colony)
+    {
+        return builderLevel >= upgradeLevel || builderLevel == 2
+                || citizen.getWorkBuilding().getID().equals(buildingLocation)
+                || isLocationTownhall(colony, buildingLocation);
+    }
+
+    private boolean isLocationTownhall(Colony colony, BlockPos buildingLocation)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/colony/workorders/WorkOrderBuild.java]: MINOR Make "isLocationTownhall" a "static" method. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2325)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 9 issues
* MAJOR 4 major
* MINOR 5 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Colony.java#L34](https://github.com/Minecolonies/minecolonies/blob/0f75be7503b0749280e6ab9a52319d4aeda0c834/src/main/java/com/minecolonies/colony/Colony.java#L34): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1200)

**Comment by Raycoms** [comment]
We maybe should analyze yesterdays crash first.

**Comment by cltnschlosser** [comment]
@Raycoms @Kostronor Can we get an issue for that crash?

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 9 issues
* MAJOR 4 major
* MINOR 5 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Colony.java#L34](https://github.com/Minecolonies/minecolonies/blob/56e26bd05fe79f176cad1dce4a9492872dadb9d8/src/main/java/com/minecolonies/colony/Colony.java#L34): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1200)

File: src/main/java/com/minecolonies/colony/Colony.java [hunk 9]
```diff
@@ -654,35 +697,67 @@ private void spawnCitizen(CitizenData data)
...
-                            EntityUtils.getPlayersFromUUID(world, permissions.getMessagePlayers()),
+                            ServerUtils.getPlayersFromUUID(world, permissions.getMessagePlayers()),
                             "tile.blockHutTownHall.messageMaxSize");
                 }
             }
 
             entity.setColony(this, citizenData);
 
-            entity.setPosition(spawnPoint.getX(), spawnPoint.getY(), spawnPoint.getZ());
+            entity.setPosition(spawnPoint.getX() + 0.5D, spawnPoint.getY() + 0.1D, spawnPoint.getZ() + 0.5D);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/colony/Colony.java]: MINOR Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)
MINOR Assign this magic number 0.1D to a well-named constant, and use the constant instead. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)
MINOR Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 12 issues
* MAJOR 4 major
* MINOR 8 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Colony.java#L34](https://github.com/Minecolonies/minecolonies/blob/b5fdc139d96e3c082a12d932e422e5eba9b98f94/src/main/java/com/minecolonies/colony/Colony.java#L34): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1200)

**Comment by Kostronor** [comment]
I suggest merging #63 and then finishing this :) @Minecolonies/maintainers

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 12 issues
* MAJOR 4 major
* MINOR 8 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Colony.java#L34](https://github.com/Minecolonies/minecolonies/blob/6573362c020019c761d59c03593ca152e72052be/src/main/java/com/minecolonies/colony/Colony.java#L34): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1200)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 12 issues
* MAJOR 4 major
* MINOR 8 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Colony.java#L34](https://github.com/Minecolonies/minecolonies/blob/61c6c8d7c8233dcb4c565f1241a31f04c980ea81/src/main/java/com/minecolonies/colony/Colony.java#L34): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1200)

**Comment by Raycoms** [comment]
:shipit:

**Pull request merged**